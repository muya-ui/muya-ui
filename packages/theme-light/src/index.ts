export { default } from './theme';
export * from './theme';
export * from './icons';
export * from './interfaces';
export * from './components/GlobalStyles';
export * from './components/FontPenrose';
export * from './components/SvgIcon';
export * from './utils';
