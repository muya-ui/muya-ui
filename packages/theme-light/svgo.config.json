{
  "plugins": [
    {
      "removeAttrs": {
        "attrs": "(class)"
      }
    },
    {
      "removeDimensions": true
    },
    {
      "cleanupAttrs": true
    }
  ]
}
