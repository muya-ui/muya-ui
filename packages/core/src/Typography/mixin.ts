import { css } from 'styled-components';

import { makeEllipsisStyle } from '../styled/mixins/ellipsis';
import { ITypographyBaseProps } from './types';
import { ITheme } from '@muya-ui/theme-light';

export const baseStyle = css`
  box-sizing: border-box;
  margin: 0;
  text-decoration: ${(props: ITypographyBaseProps) => props.delete && 'line-through'};
  text-decoration: ${(props: ITypographyBaseProps) => props.underline && 'underline'};
  ${(props: ITypographyBaseProps) => makeEllipsisStyle(props.ellipsis)};
`;

export const getFontStyle = (props: ITypographyBaseProps, theme: ITheme) => {
  const { lineHeight: lineHeightProp, fontSize: fontSizeProp } = props;

  const {
    typography,
    components: { Typography: token },
  } = theme;
  let fontSize;
  let lineHeight;

  if (typeof fontSizeProp === 'number') {
    fontSize = `${fontSizeProp}px`;
  } else {
    fontSize = `${typography.spec.fontSize[fontSizeProp || token.defaultFontLevel]}px`;
  }

  if (lineHeightProp) {
    lineHeight = lineHeightProp;
  } else if (typeof fontSizeProp === 'number') {
    lineHeight = typography.spec.global.lineHeight;
  } else {
    lineHeight = `${typography.spec.lineHeight[fontSizeProp || token.defaultFontLevel]}px`;
  }

  return {
    fontSize,
    lineHeight,
  };
};
