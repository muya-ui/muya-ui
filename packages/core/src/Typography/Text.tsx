import styled, { css } from 'styled-components';

import { withThemeForStyled } from '../utils/withTheme';
import { baseStyle, getFontStyle } from './mixin';
import { ITypographyBaseProps } from './types';

const Text = styled.span<ITypographyBaseProps>`
  ${props => {
    const { ellipsis, type, strong, color, theme } = props;
    const {
      colors,
      typography,
      components: { Typography: token },
    } = theme;
    const { lineHeight, fontSize } = getFontStyle(props, theme);

    return css`
      ${ellipsis && 'display: inline-block'};
      color: ${type
        ? colors.pattern.feature[type]
        : colors.pattern.text[color || token.defaultTextType]};
      font-weight: ${strong && typography.spec.fontWeight.semibold};
      font-size: ${fontSize};
      line-height: ${lineHeight};
      ${baseStyle}
    `;
  }}
`;

export default withThemeForStyled(Text);
