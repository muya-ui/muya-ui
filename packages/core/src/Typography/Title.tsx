import React from 'react';
import styled, { css } from 'styled-components';

import { ITypographyHeadingLevel } from '@muya-ui/theme-light';

import useTheme from '../utils/useTheme';
import { baseStyle } from './mixin';
import { ITypographyTitleProps } from './types';

export const StyledTitle = styled.div<ITypographyTitleProps>`
  ${props => {
    const {
      theme: {
        colors,
        typography,
        components: { Typography: token },
      },
      color,
      level,
    } = props;
    const fontLevel = token.titleFontLevelMap[`h${level}` as ITypographyHeadingLevel];
    return css`
      ${baseStyle}
      color: ${colors.pattern.text[color || token.defaultTitleType]};
      font-weight: ${typography.spec.fontWeight.semibold};
      font-size: ${typography.spec.fontSize[fontLevel]}px;
      line-height: ${typography.spec.lineHeight[fontLevel]}px;
      margin-bottom: ${token.titleMarginBottom};
      & + & {
        margin-top: ${token.titleMarginTop};
      }
    `;
  }}
`;

export default React.forwardRef<HTMLHeadElement, ITypographyTitleProps>((props, ref) => {
  const { level = 1 } = props;
  const asProp = `h${level}` as ITypographyHeadingLevel;
  const theme = useTheme();
  return (
    <StyledTitle
      theme={theme}
      as={asProp}
      {...props}
      level={level}
      ref={ref as React.RefObject<HTMLDivElement>}
    />
  );
});
