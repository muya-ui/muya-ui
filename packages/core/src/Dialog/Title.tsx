import React from 'react';
import styled, { css } from 'styled-components';

import { CloseIcon } from '@muya-ui/theme-light';

import IconButton from '../IconButton';
import { IThemedBaseProps } from '../types';
import Typography from '../Typography';
import { StyledTitle } from '../Typography/Title';
import memoForwardRef from '../utils/memoForwardRef';
import useTheme from '../utils/useTheme';
import { IDialogTitleProps } from './types';

const StyledDialogTitle = styled.div<IThemedBaseProps & IDialogTitleProps>`
  position: relative;
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  box-sizing: border-box;
  flex: 0 0 auto;
  ${props => {
    const token = props.theme.components.Dialog.title;
    return css`
      border-bottom: ${token.borderBottom};
      padding: ${token.paddingVertical}px ${token.paddingHorizontal}px;
      ${props.fullWidth && `padding: ${token.fullWidthPadding}`};
    `;
  }}
  ${StyledTitle} {
    margin-bottom: 0;
  }
`;

const StyledIconWrapper = styled(IconButton)`
  && {
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    right: ${props => props.theme.components.Dialog.title.paddingHorizontal}px;
    padding: 0;
  }
`;

export default memoForwardRef<HTMLDivElement, IDialogTitleProps>((props, ref) => {
  const theme = useTheme();
  const titleToken = theme.components.Dialog.title;
  const Close = titleToken.closeIcon || CloseIcon;
  const {
    children,
    onClose,
    hideClose,
    disableTypography,
    closeIcon,
    closeButtonSize,
    level = titleToken.defaultTitleLevel,
    ...other
  } = props;

  return (
    <StyledDialogTitle ref={ref} theme={theme} {...other}>
      {disableTypography ? children : <Typography.Title level={level}>{children}</Typography.Title>}
      {!hideClose && (
        <StyledIconWrapper size={closeButtonSize} theme={theme} onClick={onClose}>
          {closeIcon || <Close />}
        </StyledIconWrapper>
      )}
    </StyledDialogTitle>
  );
});
