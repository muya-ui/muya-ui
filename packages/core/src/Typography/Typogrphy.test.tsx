import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import P from './Paragraph';
import Text from './Text';
import Title from './Title';
import { getFontStyle } from './mixin';
import { muyaThemeLight } from '@muya-ui/theme-light';

test('should render correctly', () => {
  const wrapper = renderer.create(<Text>Test</Text>).toJSON();
  expect(wrapper).toMatchSnapshot();
});

test('should ellipsis', () => {
  const wrapper = mount(
    <Text ellipsis style={{ width: '1px' }}>
      TestTestTestTestTest
    </Text>,
  );
  const wrapperEllipsis = mount(<Text ellipsis={{ rows: 4 }} />);
  const wrapperEllipsis1 = mount(<Text ellipsis={{ rows: 0 }} />);
  expect(wrapper).toHaveStyleRule('text-overflow', 'ellipsis');
  expect(wrapperEllipsis).toHaveStyleRule('-webkit-line-clamp', '4');
  expect(wrapperEllipsis1).toHaveStyleRule('-webkit-line-clamp', '3');
});

test('should be strong', () => {
  const wrapper = mount(<Text strong>TestTestTestTestTest</Text>);
  const wrapperP = mount(<P strong>TestTestTestTestTest</P>);
  const wrapperTitle = mount(<Title strong>TestTestTestTestTest</Title>);
  expect(wrapper).toHaveStyleRule('font-weight', '600');
  expect(wrapperP).toHaveStyleRule('font-weight', '600');
  expect(wrapperTitle).toHaveStyleRule('font-weight', '600');
});

test('should be delete', () => {
  const wrapper = mount(<Text delete>TestTestTestTestTest</Text>);
  expect(wrapper).toHaveStyleRule('text-decoration', 'line-through');
});

test('should be underline', () => {
  const wrapper = mount(
    <Text underline ellipsis>
      TestTestTestTestTest
    </Text>,
  );
  expect(wrapper).toHaveStyleRule('text-decoration', 'underline');
});

test('should accept color and fontSize', () => {
  const wrapper = renderer
    .create(
      <Text strong color="assistant" fontSize="s6">
        TestTestTestTestTest
      </Text>,
    )
    .toJSON();
  const wrapperP = renderer
    .create(
      <P strong color="secondary" fontSize="s3">
        TestTestTestTestTest
      </P>,
    )
    .toJSON();
  expect(wrapper).toMatchSnapshot();
  expect(wrapperP).toMatchSnapshot();
});

test('getFontStyle should return correctly', () => {
  const res1 = getFontStyle(
    {
      lineHeight: '200px',
    },
    muyaThemeLight,
  );
  expect(res1.lineHeight).toBe('200px');
  expect(res1.fontSize).toBe(`${muyaThemeLight.typography.spec.fontSize.s1}px`);

  const res2 = getFontStyle({}, muyaThemeLight);
  expect(res2.lineHeight).toBe(`${muyaThemeLight.typography.spec.lineHeight.s1}px`);
  expect(res2.fontSize).toBe(`${muyaThemeLight.typography.spec.fontSize.s1}px`);

  const res3 = getFontStyle(
    {
      fontSize: 's2',
    },
    muyaThemeLight,
  );
  expect(res3.lineHeight).toBe(`${muyaThemeLight.typography.spec.lineHeight.s2}px`);
  expect(res3.fontSize).toBe(`${muyaThemeLight.typography.spec.fontSize.s2}px`);

  const res4 = getFontStyle(
    {
      fontSize: 's2',
      lineHeight: '200px',
    },
    muyaThemeLight,
  );
  expect(res4.fontSize).toBe(`${muyaThemeLight.typography.spec.fontSize.s2}px`);
  expect(res4.lineHeight).toBe('200px');

  const res5 = getFontStyle(
    {
      fontSize: 's3',
      lineHeight: 1.1,
    },
    muyaThemeLight,
  );
  expect(res5.lineHeight).toBe(1.1);
  expect(res5.fontSize).toBe(`${muyaThemeLight.typography.spec.fontSize.s3}px`);

  const res6 = getFontStyle(
    {
      fontSize: 20,
    },
    muyaThemeLight,
  );
  expect(res6.lineHeight).toBe(muyaThemeLight.typography.spec.global.lineHeight);
  expect(res6.fontSize).toBe('20px');

  const res7 = getFontStyle(
    {
      fontSize: 20,
      lineHeight: '28px',
    },
    muyaThemeLight,
  );
  expect(res7.fontSize).toBe('20px');
  expect(res7.lineHeight).toBe('28px');
});
