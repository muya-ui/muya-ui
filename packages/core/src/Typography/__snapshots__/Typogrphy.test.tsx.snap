// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should accept color and fontSize 1`] = `
.c0 {
  color: rgba(153,153,153,1);
  font-weight: 600;
  font-size: 24px;
  line-height: 32px;
  box-sizing: border-box;
  margin: 0;
}

<span
  className="c0"
  color="assistant"
  fontSize="s6"
>
  TestTestTestTestTest
</span>
`;

exports[`should accept color and fontSize 2`] = `
.c0 {
  padding: 0;
  color: rgba(187,187,187,1);
  font-weight: 600;
  font-size: 16px;
  line-height: 22px;
  box-sizing: border-box;
  margin: 0;
}

<p
  className="c0"
  color="secondary"
  fontSize="s3"
>
  TestTestTestTestTest
</p>
`;

exports[`should render correctly 1`] = `
.c0 {
  color: rgba(102,102,102,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

<span
  className="c0"
>
  Test
</span>
`;
